src/Extensions/Microsoft.Diagnostics.Monitoring.Extension.Common/EgressHelper.cs

```diff
@@ -35,8 +37,24 @@ private static async Task<int> Egress<TOptions>(EgressProvider<TOptions> provide
             EgressArtifactResult result = new();
             try
             {
-                string jsonConfig = Console.ReadLine();
-                ExtensionEgressPayload configPayload = JsonSerializer.Deserialize<ExtensionEgressPayload>(jsonConfig);
+                StdInStream = Console.OpenStandardInput();
+
+                var payloadVersionBuffer = new byte[sizeof(int)];
+                await ReadExactlyAsync(payloadVersionBuffer, token);
```

Comment:
Try taking a look at using [BinaryReader](https://learn.microsoft.com/en-us/dotnet/api/system.io.binaryreader?view=net-8.0).
- It seems to only read exactly what is asked for from the stream.
- It establishes that values are written in little endian form (whereas right now we leave that up to the runtime).
- We avoid having to allocate buffers for each piece of data.

This will require you to use [BinaryWriter](https://learn.microsoft.com/en-us/dotnet/api/system.io.binarywriter?view=net-8.0) on the dotnet-monitor side.